### ISSUE TITLE
.resize() crops image when crop is unnecessary

### ISSUE THREAD

**Issue by suprMax** [lovell/sharp#831]
I'm having trouble with resize function cropping sides of the image when only resize is needed:

```
// assume expressjs middleware
const resizeTransform = sharp().resize(/* either numbers or undefined-s from querystring */);
const downloadStream = request(req.query.src);
req.pipe(downloadStream).pipe(resizeTransform).pipe(res);
```

passing along width=200 produces this image:
![image](https://cloud.githubusercontent.com/assets/789031/26588402/85045480-4555-11e7-99be-3b08792fdd67.png)

However passing 199 or 201 produces expected result:
![image-2](https://cloud.githubusercontent.com/assets/789031/26589059/48da681c-4557-11e7-9a0e-457bd996924c.png)

original:
![arnold-transformation-challenge-1-600x399-1](https://cloud.githubusercontent.com/assets/789031/26588412/902d413c-4555-11e7-8f1c-3ae886393645.png)

Seems like some strange bug to me. I was expecting to always get a downscaled image in the original aspect ratio. I couldn't find anything in the documentation regarding controlling the crop. How does one achieves resizing this without cropping, and only having crop when both width and height are present?

I already tried converting to JPEG first, passing both dimensions and playing with .max, .min methods, no luck.

Running node v6.10.2, sharp v0.18.0

**Comment by lovell** [comment]
Hello, I'm currently investigating a problem with JPEG shrink-on-load that appears to have been introduced in v0.18.0 and I believe this is related. v0.17.3 works as expected.

**Comment by suprMax** [comment]
@lovell ah thank you, I will try downgrading to v0.17.3 and check

**Comment by lovell** [comment]
Commit 9e39a7f fixes the problem I've been investigating. Are you able to confirm the latest code on the `master` branch fixes things for you too?

**Comment by lovell** [comment]
...and commit f42a1ce fixes it properly, adding your test case too, thanks for reporting this.

**Comment by lovell** [comment]
These fixes are in v0.18.1 now available via npm. Please re-open if you're still having related problems.

**Comment by suprMax** [comment]
Just checked with v0.18.1 and it works great!
